File: go/task-tracker/internal/task/storage.go
package task

import (
	"encoding/json"
	"os"
	"path/filepath"
)

const tasksFileName = "tasks.json"

// GetTasksFilePath returns the path to the tasks.json file in the current directory.
func GetTasksFilePath() string {
	return filepath.Join(".", tasksFileName) // "." specifies the current directory
}

// ReadTasks reads tasks from the JSON file.
func ReadTasks() ([]Task, error) {
	filePath := GetTasksFilePath()
	if _, err := os.Stat(filePath); os.IsNotExist(err) {
		return []Task{}, nil
	}
	data, err := os.ReadFile(filePath)
	if err != nil {
		return nil, err
	}
	var tasks []Task
	if err := json.Unmarshal(data, &tasks); err != nil {
		return nil, err
	}
	return tasks, nil
}

// WriteTasks writes tasks to the JSON file.
func WriteTasks(tasks []Task) error {
	filePath := GetTasksFilePath()
	data, err := json.MarshalIndent(tasks, "", "  ")
	if err != nil {
		return err
	}
	if err := os.WriteFile(filePath, data, 0644); err != nil {
		return err
	}
	return nil
}

File: python/task-tracker/functionality/storage.py
import json
import os
from task import Task

TASKS_FILE = "tasks.json"

def read_tasks():
    if not os.path.exists(TASKS_FILE):
        return []
    with open(TASKS_FILE, "r") as f:
        try:
            data = json.load(f)
            return [Task.from_dict(task) for task in data]
        except json.JSONDecodeError:
            return []

def write_tasks(tasks):
    with open(TASKS_FILE, "w") as f:
        json.dump([task.to_dict() for task in tasks], f, indent=2)
